src/autocomplete/databases/mongo/mongo-extract-commands.ts

```diff
@@ -303,10 +332,26 @@ export interface UnexpectedError {
     message: unknown;
 }
 
+type ExtractionError = ParsingError | UnexpectedError;
+
+interface CommandParsingResult {
+    command: Command;
+    error?: undefined;
+}
+
+interface CommandParsingError {
+    command?: undefined;
+    error: ExtractionError;
+}
+
 export type ExtractMongoCommandsFromQueryResult =
-    | {commands: Command[]}
-    | {parseSyntaxErrors: ParserSyntaxError[]}
-    | {parseCommandsError: ParsingError | UnexpectedError};
+    | {commands: Command[]; parseSyntaxErrors?: undefined; parseCommandsError?: undefined}
+    | {parseSyntaxErrors: ParserSyntaxError[]; parseCommandsError?: undefined; commands?: undefined}
+    | {
+          parseCommandsError: ExtractionError;
+          parseSyntaxErrors?: undefined;
+          commands?: undefined;
```

Comment:
This way we can make condition blocks without annoying `if ('parseCommandsError' in command) {...}`
now we can check value existence like this `if (command.parseCommandError) {...}`